src/Hub.cpp

```diff
@@ -289,7 +309,42 @@ const sc2::Point3D* Hub::GetNextExpansion() {
         return nullptr;
 
     it->owner = Owner::CONTESTED;
-    return &(it->town_hall_location);
+    return &(*it);
+}
+
+Expansion* Hub::GetExpansionOfTownhall(const sc2::Unit& unit_) {
+    for (auto& i : m_expansions) {
+        if (std::floor(i.town_hall_location.x) == std::floor(unit_.pos.x) &&
+            std::floor(i.town_hall_location.y) == std::floor(unit_.pos.y)) {
+            return &i;
+        }
+    }
+
+    return nullptr;
+}
+
+void Hub::CapturedExpansion(const sc2::Unit& unit_) {
```

Comment:
Style: should be verb here as we do an action.

Also, I think we shouldn't really care about `if (!expansion)` condition here. All expansions should be in the list, if we missed something, I would prefer to break immediately as this looks like a bug (definitely, current expansion detection logic doesn't work on some maps).

Here and in the other places.